### ISSUE TITLE
No documentation about the purpose/usage of kParseNumbersAsStringsFlag and the RawNumber reader handler.

### ISSUE THREAD

**Issue by comand** [miloyip/rapidjson#581]
Support for RawNumber was added recently, and the way I found out was because my reader handlers no longer compiled. I checked the commits, and there doesn't appear to be any rationale for why one would enable this flag, nor are there any comments or doc in the code that would guide me. It would be great if I didn't have to deal with this at all, and I wouldn't have to implement a method in my handler that just returns false, if I don't intend to use the feature.

However, a bit more information about what this is, and why I might want it, would help me to explain to others that are looking at my code.

Thanks in advance,
- C.

**Comment by miloyip** [comment]
Sorry that this has broken your code.

To make it easier to adapt to future enhancement, it will be better to let handlers inherit from `BaseReaderHandler<T>`. For example,

~~~
struct MyHandler : public rapidjson::BaseReaderHandler<MyHandler> {
    // ...
};
~~~

**Comment by comand** [comment]
I'll take another look at that -- I looked at it last week to see if it made it clearer about what to return from my RawNumber handler if I didn't intend to support RawNumber.

**Comment by miloyip** [comment]
I will try to make the RawNumber() not breaking existing code.

**Comment by corporateshark** [comment]
@comand `RawNumber()` is never invoked if no `kParseNumbersAsStringsFlag` is specified. It does not matter what to return.

**Comment by comand** [comment]
I derived my handler from BaseReaderHandler, and now I need to be even less concerned about what gets returned by this function. Thanks for the pointer to the motivating issue.

**Comment by miloyip** [comment]
Fix in f4ea0d3f640fc29e857cc2f187cef75bc2a035a5

**Comment by comand** [comment]
Thanks!
